modules/auxiliary/gather/enum_azuresubdomains.rb

```diff
@@ -0,0 +1,155 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+    require 'resolv'
+    include Msf::Exploit::Remote::DNS::Enumeration
+
+    def initialize(info = {})
+      super(
+        update_info(
+          info,
+          'Name' => 'Azure Subdomain Scanner and Enumerator',
+          'Description' => 'This module can be used for enumerating public Azure services by locating valid subdomains through various DNS queries.',
+          'Author' => ['RoseSecurity <RoseSecurityConsulting[at]protonmail.me>'],
+          'License' => MSF_LICENSE,
+        )
+      )
+    register_options(
+        [
+          OptString.new('DOMAIN', [true, 'The target domain without TLD (Ex: victim rather than victim.org)']),
+          OptBool.new('PERMUTATIONS', [false, 'Prepend and append permutated keywords to domain (This option can take minutes to complete)', false]),
+          OptBool.new('ENUM_A', [true, 'Enumerate DNS A record', true]),
+          OptBool.new('ENUM_CNAME', [true, 'Enumerate DNS CNAME record', true]),
+          OptBool.new('ENUM_MX', [true, 'Enumerate DNS MX record', true]),
+          OptBool.new('ENUM_NS', [true, 'Enumerate DNS NS record', true]),
+          OptBool.new('ENUM_SOA', [true, 'Enumerate DNS SOA record', true]),
+          OptBool.new('ENUM_TXT', [true, 'Enumerate DNS TXT record', true]),
+        ])
+    end
+
+    def run
+      # Array of subdomains to enumerate
+      domain = datastore['DOMAIN']
+      subdomains = [ ".onmicrosoft.com",
```

Comment:
Did this list come from somewhere? If so, please add a comment with a reference URL, or add the URL to the `References` module `info` Hash.